app/lib/providers/authentication/authentication.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:app/common/enums/app_state.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+
+class AuthenticationProvider extends ChangeNotifier {
+  FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
```

Comment:
- [ ] Are you sure this is a deep copy? That we're not making another instance?

Reply:
We need an instance of firebase in order to use its method like signInWithEmailAndPassword. We won't be able to use these firebase method without creating an instance of firebaseauth. Unless you know other method, please let me know.

Reply:
I thought FirebaseAuth.instance was a static item, but [these examples](https://firebase.flutter.dev/docs/auth/usage/) do what you do. No changes needed.